### ISSUE TITLE
TypeScript definitions for PluginContext are Incorrect

### ISSUE THREAD

**Issue by wangweixuan** [rollup/rollup#2974]
- Rollup Version: 1.16.3
- Operating System (or Browser): not concerned
- Node Version: not concerned

### How Do We Reproduce?

I want to write a plugin in TypeScript, for example:
```typescript
import { PluginImpl } from 'rollup'

export const myPlugin: PluginImpl<{ }> = () => {
  return {
    name: 'my-plugin',

    transform(code, id) {
      this.error('There is something wrong.', 42) // Here
      return undefined
    }
  }
}
```

### Expected Behavior

The code compiles and runs. The plugin reports an error at the 42<sup>nd</sup> character position in any file.

### Actual Behavior

The code won't compile. TypeScript complains that `Argument of type '42' is not assignable to parameter of type '{ column: number; line: number; }'.`

### What's Wrong

In Rollup's type definition file:
https://github.com/rollup/rollup/blob/17394593810eb4011fd8bd8285cd1fe4f6ee7f4b/src/rollup/types.d.ts#L125
https://github.com/rollup/rollup/blob/17394593810eb4011fd8bd8285cd1fe4f6ee7f4b/src/rollup/types.d.ts#L149

While in the document:
https://github.com/rollup/rollup/blob/master/docs/05-plugin-development.md#thiserrorerror-string--error-position-number--void
```typescript
this.error(error: string | Error, position?: number) => void
```
https://github.com/rollup/rollup/blob/master/docs/05-plugin-development.md#thiswarnwarning-string--rollupwarning-position-number-
```typescript
this.warn(warning: string | RollupWarning, position?: number)
```

**Comment by lukastaegert** [comment]
Thanks for spotting this! As a matter of fact, both should work—at least there are tests for them. I will have a look at definitions and docs.

**Comment by lukastaegert** [comment]
Fix at #2975
